### DISCUSSION TITLE
Best-practice theming structure with @chakra-ui/gatsby-plugin

### DISCUSSION THREAD

**Discussion by calebplum** [chakra-ui/chakra-ui#8893] [Q&A]
### Description

I'm using @chakra-ui/gatsby-plugin and I'd like to implement the recommended theme structure described in the Chakra UI documentation [here](https://chakra-ui.com/docs/styled-system/customize-theme#scaling-out-your-project).

However, the Gatsby plugin expects to find the theme entrypoint file `src/@chakra-ui/gatsby-plugin/theme.js` ([reference](https://www.gatsbyjs.com/plugins/@chakra-ui/gatsby-plugin/#customizing-the-theme)). I can't find anywhere in the plugin documentation that describes how to specify a custom location.

Is it possible to specify a custom location for the theme entrypoint file in the Gatsby plugin?

### Chakra UI Version

@chakra-ui/react 2.7.0, @chakra-ui/gatsby-plugin 3.1.3

### Browser

Firefox

### Operating System

- [ ] macOS
- [ ] Windows
- [X] Linux

**Comment by segunadebayo** [comment]
cc @TylerAPfledderer Can you help with this?

**Comment by TylerAPfledderer** [comment]
Thanks, @segunadebayo!

Hey @calebplum!

The setup in the plugin should work for the custom structure provided in our docs. 

Per the Gatsby Docs for [Theme Shadowing](https://www.gatsbyjs.com/docs/how-to/plugins-and-themes/shadowing/) your entry point has to match the directory structure provided by the plugin. 

Also we do not have the `theme` file exported strictly as a `*.js` file, which means `theme.{js | ts}` or `theme/index.{js | ts}` should work here.

So your file directory path has to be `src/@chakra-ui/gatsyby-plugin/theme` and be able to use a structure like:
```
|-- src
    |-- @chakra-ui
        |-- gatsby-plugin
            |-- theme
                |-- index.ts
                |-- foundations
                |-- styles.ts
                |-- semanticTokens.ts
```

I can certainly look at seeing if the docs can be further clarified without creating any other confusion 😄

**Comment by calebplum** [comment]
Hi @TylerAPfledderer! Many thanks for your helpful response.

I've implemented your suggestion but none of the styles in my main theme entrypoint are taking effect with the directory structure you described.

This is what my project's directory structure looks like:
```
src
├── @chakra-ui
│   └── gatsby-plugin
│       └── theme
│           └── index.js
 ...
```

The contents of my `src/@chakra-ui/gatsby-plugin/theme/index.js` is as follows (just some basic styles for testing purposes):

```js
import { extendTheme } from "@chakra-ui/react"
const theme = {
  styles: {
    global: {
      body: {
        bg: 'orange',
        color: 'red',
      },
      b: {
        color: 'teal.500',
        _hover: {
          textDecoration: 'underline'
        }
      },
      div: {
        bg: 'yellow'
      }
    }
  },
  colors: {
    primary: "rebeccapurple",
  },
}

export default extendTheme(theme)
```

When I move this file to `src/@chakra-ui/gatsby-plugin/theme.js` it works, but it doesn't work when it's placed at `src/@chakra-ui/gatsby-plugin/theme/index.js`.

I've cleaned my project directory before starting the development server with `npm run clean && npm run develop` but no luck. There are also no errors in the development server's output or the web console.

Can you see anything wrong my configuration, or do you know how I can troubleshoot this further?

Many thanks!
